### DISCUSSION TITLE
What is the difference between filer.sync and filer.backup ?

### DISCUSSION THREAD

**Discussion by LuuucasChen** [seaweedfs/seaweedfs#4707] [Q&A]
The two functions look similar

**Comment by kmlebedev** [comment]
`filer.backup` used events from internal logs
`filer.sync` used events from the queue

**Reply by LuuucasChen** [reply]
Does that mean they achieve similar results? I want to use SeaweedFS to build a file system that requires real-time synchronization within the same city but in difference data centre. Which feature would be more suitable?

**Reply by LuuucasChen** [reply]
Thank you for your response. And it seems that backup is queue-based, while sync is log-based.

**Comment by kmlebedev** [comment]
There is also `filer.replicate` is queue-based  to replicates events on external disk that works similarly to `filer.backup` is log-based on local disk. And `filer.sync` is replicates events to another filler

**Reply by chrislusf** [reply]
Sync is two-direction. Backup is one-direction.

**Reply by LuuucasChen** [reply]
Thank you. There is another issue to address. When using sync or backup to synchronize or back up files, the file content remains the same, but the fid changes. In the event of a source cluster/node failure and the need to switch to a backup node, the file's fid has already changed to that of the backup file, making it difficult to retrieve the file. Do you have any suggestions for this issue? Should we use the file directory as the storage for retrieval information?
